### ISSUE TITLE
elastic-tube-1d:  C++ results are not consistent with python

### ISSUE THREAD

**Issue by carme-hp** [precice/tutorials#195]
The results obtained with the python and c++ versions differ.  The python version is the one believed to be correct. The difference in the results originates from the fluid solver (fluid-cpp and solid-cpp lead to the same results than fluid-cpp and solid-python) . The difference in the results can be observed by naked eye after plotting with 
```
python3 postproc/fluid.py diameter postproc/cpp_fluid_
```
![cpp](https://user-images.githubusercontent.com/71499004/114310578-78d2b900-9aeb-11eb-994d-8da0bd021674.png)
for the cpp results and 
```
python3 postproc/fluid.py diameter postproc/out_fluid_
```
for the python results. 
![python](https://user-images.githubusercontent.com/71499004/114310579-796b4f80-9aeb-11eb-9c58-7b6752e3917e.png)
You can see that for the python tube the diameter is smaller at the end of the tube. This is the expected behavior, as we are further away from the inlet. 

I have tried to debug this issue unsuccessfully so far. For anyone who wants to try: 
- `alpha` and `dx` don't have the same value in each version. `alpha=0` in python but `alpha=50` in cpp, whereas `dx=0.1` in python but `dx=0.001` in cpp .
- If we use the python's values of `alpha` and `dx`, then the cpp does not converge. If we print values we see that the first two iterations look like in python but then pressure explodes.  
- The most likely explanation for that would be some typo in a sign but I couldn't find it. 

Finally, if someone manages to fix this issue, please make sure to adress the [parallel issue](https://github.com/precice/elastictube1d/issues/40#issue-855395710) as well!

**Comment by fsimonis** [comment]
@carme-hp Thank you for investing so much work into this.
I spend the last 2 days debugging this and I finally found the issue.

The reason why the issue is so difficult to find is that the python and the cpp version of the fluid solver have different interface and the computation of some values were spread over multiple files.

In a nutshell:
* The cpp version used `t` to compute the inlet velocity, while the python version used `t+dt`.
* The cpp version used `dx = 1.0 / (N * kappa)` while the python version used `dx = L / 100 ` (100 being kappa)

Influenced LHS and Res from iteration 0.

The state after some "fixes":

![image](https://user-images.githubusercontent.com/13552216/114688935-7eabe280-9d15-11eb-94a0-f8daefe3875e.png)

**Comment by carme-hp** [comment]
@fsimonis I'm very happy it is finally solved! thanks:)

**Comment by MakisH** [comment]
Closed by #197.
